src/posit/connect/client.py

```diff
@@ -192,10 +197,54 @@ def with_user_session_token(self, token: str) -> Client:
         Client
             A new Client instance authenticated with an API key exchanged for the user session token.
 
+        Raises
+        ------
+        ValueError
+            If the provided token is `None` or empty or if the exchange response is malformed.
+        ClientError
+            If the token exchange request with the Connect Server fails.
+
         Examples
         --------
         >>> from posit.connect import Client
         >>> client = Client().with_user_session_token("my-user-session-token")
+
+        Example using user session token from Shiny session:
+        >>> from posit.connect import Client
+        >>> from shiny.express import render, session
+        >>>
+        >>> client = Client()
+        >>>
+        >>> @reactive.calc
+        >>> def visitor_client():
+        ...     ## read the user session token and generate a new client
+        ...     user_session_token = session.http_conn.headers.get(
+        ...         "Posit-Connect-User-Session-Token"
+        ...     )
+        ...     return client.with_user_session_token(user_session_token)
+        >>> @render.text
+        >>> def user_profile():
+        ...     # fetch the viewer's profile information
+        ...     return visitor_client().me
+
+        Example with fallback when header is missing in API request:
+        >>> from posit.connect import Client
+        >>> import requests
+        >>>
+        >>> def get_client(request):
+        ...     base_client = Client()
+        ...     token = request.headers.get("Posit-Connect-User-Session-Token")
+        ...     if token:
+        ...         return base_client.with_user_session_token(token)
+        ...     else:
+        ...         return None
```

Comment:
```suggestion
        ...     # It's important to make catch if the visitor's token could not be retrieved (for example, if this app allows unauthenticated access) and handle that in cases where a token is expected (see below)

        ...     token = request.headers.get("Posit-Connect-User-Session-Token")
        ...     if token:
        ...         return base_client.with_user_session_token(token)
        ...     else:
        ...         return None
```

I'm not sure this is the right place for it, but having some text that explains why / when this is important would be nice. It also makes me think: maybe we need to think about a nicer way to not require there to be two `if`/`else` or `try`/`catch` blocks for something like this. We don't have to perfect it here, but this to me is a sign we might rethink how we tell folks to do this and see if there's a cleaner way we can provide a nice experience for someone using this feature.

Reply:
Yeah, letting the use cases drive the design a bit...

I think the ideal is escaping the "wither" pattern and forcing client creation using either the API Key or the Session Token. But that isn't possible at the moment.

Reply:
_nods_ yeah, and it's not _just_ the API Key or Session token theoretically. In the oauth world it might be: visitor Oauthcreds, then service account creds, then something that catches and fails gracefully. We won't solve that entirely here, but that's the experience we should aim to provide if we can

Reply:
agreed! will simplify the example for now. can circle back if/when the visitor api key feature gets traction and we continue to evolve it.